app/models/bank_transaction.rb

```diff
@@ -33,4 +33,55 @@ class BankTransaction < ActiveRecord::Base
   def image_url
     'data:image/png;base64,' + Base64.encode64(self.image)
   end
+
+  def assign_to_invoice
+    return false unless supplier
+
+    content = text
+    content += "\n" + reference if reference.present?
+    invoices = supplier.invoices.unpaid.select {|i| content.include? i.number}
+    invoices_sum = invoices.inject(0) {|sum,i| sum + i.amount }
+    return false if amount != -invoices_sum
+
+    ActiveRecord::Base.transaction do
+      link = FinancialLink.new
+      invoices.each {|i| i.update_attributes financial_link: link, paid_on: date}
+      update_attribute :financial_link, link
+    end
+
+    return true
+  end
+
+  def assign_to_ordergroup
+    m = /FS(?<group>\d+)(\.(?<user>\d+))?(?<parts>([A-Za-z]+\d+(\.\d+)?)+)/.match(reference)
+    return unless m
```

Comment:
My hunch here would be to put this and further matching elsewhere (to avoid mega-models). Still this is pretty coupled to a bank transaction. I don't have a clear head on what is needed here to keep a clean architecture. ... :/

Reply:
Perhaps something that takes bank transactions, and deducts what they are for.
This needs knowledge of other models (like you're querying `Ordergroup` here).

Thinking aloud, perhaps something like this: with input a transaction with reference `FS20.5.Membership25` could result in the hash
```ruby
{
  ordergroup_id: 20,
  user_id: 5,
  financial_transactions: [
    { name_short: 'Membership' amount: 25 }
  ]
}
```

This method would do all the validation it can (except checking that `name_short` exists, and that user and ordergroup would belong together, if that may ever become a requirement).
This could also help to aid the user in interpreting the result: e.g. when the amounts don't add up, the ordergroup could choose in Foodsoft how to distribute the funds over transaction classes (or if we'd start looking at unmatched transactions from the user's verified IBAN, then he could also correct ordergroup typo's).

I hope you don't experience my thoughts as "blocking", and will see the beautify of thinking together how this can be implemented in an extensible way. We could also just merge it, and see what happens, but perhaps we can learn something from pausing for a minute.

Reply:
i want your ideas on this, otherwise i'd just merge it into master ;-)
i'll think about it a little bit more together with the comments in the other issue and post a new PR in the next days

Reply:
i moved it into a new class